src/pages/Home.tsx

```diff
@@ -45,34 +45,22 @@ export default function Home() {
   const [selectedTags, setSelectedTags] = useState<Array<Tag>>([]);
   const [searchParams, setSearchParams] = useSearchParams();
 
-  const loadData = useCallback(async () => {
-    setIsLoading(true);
-
-    const channelsList = await apiGet<Channel[]>(endpoints.channels.url);
-    const tagsList = await apiGet<Tag[]>(endpoints.tags.url);
-    const vodsList = await apiGet<Channel[]>(endpoints.vods.url);
-
+  const loadData = useCallback(async (loader: typeof setIsLoading) => {
+    loader(true)
```

Comment:
Since it is not a pure function because we're still using `setChannels`, `setTags`, and `setVods`, I believe it is not a good thing to pass the `setIsLoading` as param and maintain it explicit.

Reply:
i dont get it, isn't the setter function supposed to be stable and not change over rerenders?

other idea to not duplicate logic would be like this, let me know what you think of this approach
```js
  const [isFetching, setIsFetching] = useState<boolean>(true);
  const hasData = !!channels.length || !!tags.length || !!vods.length;
  const isLoading = isFetching && !hasData;
  const isRefetching = isLoading && hasData;

  const loadData = useCallback(async () => {
     setIsFetching(true);
     ...

     setIsFetching(false);
  }, [apiGet]);

```

Reply:
I'm talking about passing `setIsLoading` as a param. 

The version you proposed passing setIsLoading/Fetching inside the function would make sense too.